types/authorize.go

```diff
@@ -0,0 +1,92 @@
+package types
+
+import "fmt"
+
+// A Request is the Principal, Action, Resource, and Context portion of an
+// authorization request.
+type Request struct {
+	Principal EntityUID `json:"principal"`
+	Action    EntityUID `json:"action"`
+	Resource  EntityUID `json:"resource"`
+	Context   Record    `json:"context"`
+}
+
+// A Decision is the result of the authorization.
+type Decision bool
+
+// Each authorization results in one of these Decisions.
+const (
+	Allow = Decision(true)
+	Deny  = Decision(false)
+)
+
+func (a Decision) String() string {
+	if a {
+		return "allow"
+	}
+	return "deny"
+}
+
+func (a Decision) MarshalJSON() ([]byte, error) { return []byte(`"` + a.String() + `"`), nil }
+
+func (a *Decision) UnmarshalJSON(b []byte) error {
+	*a = string(b) == `"allow"`
+	return nil
+}
+
+// A Diagnostic details the errors and reasons for an authorization decision.
+type Diagnostic struct {
+	Reasons []DiagnosticReason `json:"reasons,omitempty"`
+	Errors  []DiagnosticError  `json:"errors,omitempty"`
+}
+
+// A DiagnosticReason details the Policy index within a PolicySet, and the Position within
```

Comment:
I realize this code was just moved from elsewhere, but would it more correct to change "Policy index" to "Policy identifier" or "PolicyID" since I think it is no longer an index?